Userland/Libraries/LibCrypto/ASN1/PEM.cpp

```diff
@@ -56,4 +57,48 @@ ByteBuffer decode_pem(ReadonlyBytes data)
     return decoded;
 }
 
+Vector<ByteBuffer> decode_pems(ReadonlyBytes data)
+{
+    GenericLexer lexer { data };
+    ByteBuffer decoded;
+    Vector<ByteBuffer> certs;
+
+    enum {
+        Junk,
+        Parsing,
+    } state { Junk };
+    while (!lexer.is_eof()) {
+        switch (state) {
+        case Junk:
+            if (lexer.consume_specific("-----BEGIN"))
+                state = Parsing;
+            lexer.consume_line();
+            break;
+        case Parsing: {
+            if (lexer.consume_specific("-----END")) {
+                state = Junk;
```

Comment:
not sure what file format says… shouldnʼt you accept only empty lines as a junk? 🤔

Reply:
File format says: Certs go between BEGIN and END. Ignore the rest (there is quite a lot to ignore)